moto/iam/models.py

```diff
@@ -1093,19 +1097,29 @@ def __init__(
         self.status = status
         self.create_date = utcnow()
 
+        self._last_used: Optional[AccessKeyLastUsed] = None
+        self.role_arn: Optional[str] = None
+
+    @property
+    def last_used(self) -> Optional[AccessKeyLastUsed]:
+        return self._last_used
+
+    @last_used.setter
+    def last_used(self, value: Union[datetime, AccessKeyLastUsed, None]) -> None:
         # Some users will set this field manually
         # And they will be setting this value to a `datetime`
         # https://github.com/getmoto/moto/issues/5927#issuecomment-1738188283
-        #
-        # The `to_csv` method calls `last_used.strptime`, which currently works on both AccessKeyLastUsed and datetime
```

Comment:
this comment isn't technically made obsolete by my change here, but the importance of it is lessened (imo) and I wasn't sure where it would go. I can put it (or something like it) back in, I was waffling on that.